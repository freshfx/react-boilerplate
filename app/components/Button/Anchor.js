import styled from '@emotion/styled'

import buttonStyles from './buttonStyles'

const Anchor = styled.a`
  ${buttonStyles};
`

export default Anchor
