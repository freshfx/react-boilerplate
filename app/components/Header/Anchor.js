import styled from 'styled-components'
import NormalA from 'components/Anchor'

const Anchor = styled(NormalA)`
  padding: 2em 0;
`

export default Anchor
