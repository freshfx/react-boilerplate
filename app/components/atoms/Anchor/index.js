/**
 * A link to a certain page, an anchor tag
 */

import styled from '@emotion/styled'

const Anchor = styled.a`
  color: #41addd;

  &:hover {
    color: #6cc0e5;
  }
`

export default Anchor
