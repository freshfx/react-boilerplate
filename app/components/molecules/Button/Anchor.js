import styled from '@emotion/styled'

import buttonStyles from 'components/atoms/Button/buttonStyles'

const Anchor = styled.a`
  ${buttonStyles};
`

export default Anchor
