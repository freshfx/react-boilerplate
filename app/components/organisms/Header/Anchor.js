import styled from '@emotion/styled'

import NormalA from 'components/atoms/Anchor'

const Anchor = styled(NormalA)`
  padding: 2em 0;
`

export default Anchor
