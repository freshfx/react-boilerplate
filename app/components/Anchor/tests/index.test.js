/**
 * Testing our link component
 */

import 'jest-styled-components'
import React from 'react'
import {mount} from 'enzyme'

import Anchor from '../index'

const href = 'http://mxstbr.com/'
const renderComponent = (props = {}) =>
  mount(<Anchor href={href} {...props}>Test</Anchor>)

describe('<Anchor />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderComponent()
    expect(renderedComponent).toMatchSnapshot()
  })
})
