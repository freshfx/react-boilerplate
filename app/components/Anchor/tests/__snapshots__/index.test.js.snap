// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> should match snapshot 1`] = `
.c0 {
  color: #41addd;
}

.c0:hover {
  color: #6cc0e5;
}

<Anchor
  href="http://mxstbr.com/"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Anchor-aulrtc-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "color:#41addd;&:hover{color:#6cc0e5;}",
          ],
        },
        "displayName": "Anchor",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Anchor-aulrtc-0",
        "target": "a",
        "toString": [Function],
        "usesTheme": false,
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    href="http://mxstbr.com/"
  >
    <a
      className="c0"
      href="http://mxstbr.com/"
    >
      Test
    </a>
  </StyledComponent>
</Anchor>
`;
